pkg/operator/collection.go

```diff
@@ -503,6 +503,42 @@ func (r *collectionReconciler) makeCollectorConfig(ctx context.Context, spec *mo
 			continue
 		}
 		cfg.ScrapeConfigs = append(cfg.ScrapeConfigs, cfgs...)
+
+		change, err := nm.Status.SetMonitoringCondition(nm.GetGeneration(), metav1.Now(), cond)
+		if err != nil {
+			// Log an error but let operator continue to avoid getting stuck
+			// on a potential bad resource.
+			logger.Error(err, "setting nodemonitoring status state")
+		}
+
+		if change {
+			// convert ScrapeNodeEndpoint to ScrapeEndpoint.
+			var scrapeEndpoints []monitoringv1.ScrapeEndpoint
```

Comment:
Do you have any better ideas on how to use NodeMonitoringCRD and PodMonitoringCRD in statusUpdates?